src/coreclr/pal/src/exception/machexception.cpp

```diff
@@ -681,9 +682,24 @@ HijackFaultingThread(
     BuildExceptionRecord(exceptionInfo, &exceptionRecord);
 
 #if defined(HOST_AMD64)
+    int cpuFeatures = minipal_getcpufeatures();
```

Comment:
`minipal_getcpufeatures` is not a cheap method. I am not sure whether it is ok to call it for each hijack.

Reply:
This call and all of the related checks are not necessary. The `CONTEXT_GetThreadContextFromThreadState` can be passed the x86_AVX512_STATE instead of x86_FLOAT_STATE unconditionally and it will return AVX512, AVX or FLOAT state depending on what's available on the current thread. It will also update the `XStateFeaturesMask` on the context based on that.